decoder.go

```diff
@@ -73,7 +73,7 @@ func convertSignedInts(src interface{}, dst *reflect.Value) error {
 			return nil
 		}
 	case uintptr:
-		if t <= math.MaxInt64 && !dst.OverflowInt(int64(t)) {
+		if int64(t) <= math.MaxInt64 && !dst.OverflowInt(int64(t)) {
```

Comment:
Why do you need both `<= math.MaxInt64` and `dst.OverflowInt` check?

Reply:
The `dst` could represent any bit sized integer, and `Value.SetInt` will use [conversion](https://golang.org/src/reflect/value.go?s=42937:42967#L1426) instead of reporting overflows.

Reply:
`int64(t)` can only ever be a value that is `math.MinInt64 <= t <= math.MaxInt64` (by definition, since you're casting it to an `int64`), so the check `int64(t) <= math.MaxInt64` will *always* be true.

Reply:
Yeah :) I am trying to find a max of uintptr, but can't think of any reasonable approach. Any suggestions?

Reply:
You have a `uintptr`, and you want to verify it's safe to write in to a signed integer right?

If you convert to a `int64`, and it's negative, that means the value is too large for any signed integer. If it's positive, but too large for the specific signed integer type, `OverflowInt` will catch it.

So I think what you want is:
```
if int64(t) >= 0 && !dst.OverflowInt(int64(t)) {
```

Reply:
Well, it is a signed integer here, but an unsigned integer below. Probably, we can assume, that uintptr is always smaller than math.MaxUInt64 and report an error to the user otherwise.

Reply:
I decided to keep it simple and use keep uintptr in the same bay as other ints.